exchange/index_test.go

```diff
@@ -574,3 +576,144 @@ func TestLoadInterest(t *testing.T) {
 		require.Equal(t, reflist2[0].PayloadCID, k.PayloadCID)
 	}
 }
+
+func TestGC(t *testing.T) {
+	ctx := context.Background()
+
+	mn := mocknet.New(ctx)
+	n := testutil.NewTestNode(mn, t)
+
+	opts := Options{
+		Blockstore: n.Bs,
+		MultiStore: n.Ms,
+		RepoPath:   n.DTTmpDir,
+	}
+	exch, err := New(ctx, n.Host, n.Ds, opts)
+	require.NoError(t, err)
+
+	// generate random block1
+	blk1 := n.CreateRandomBlock(t, n.Bs)
+	require.NoError(t, exch.Index().Bstore().Put(blk1))
+
+	// generate random block2
+	blk2 := n.CreateRandomBlock(t, n.Bs)
+	require.NoError(t, exch.Index().Bstore().Put(blk2))
+
+	// set blk1 ref1 in index
+	require.NoError(t, exch.Index().SetRef(&DataRef{
+		PayloadCID:  blk1.Cid(),
+		PayloadSize: int64(len(blk1.RawData())),
+	}))
+
+	// set blk2 ref2 in index
+	require.NoError(t, exch.Index().SetRef(&DataRef{
+		PayloadCID:  blk2.Cid(),
+		PayloadSize: int64(len(blk2.RawData())),
+	}))
+
+	// check if refs are in index
+	ref1, err := exch.Index().GetRef(blk1.Cid())
+	require.NoError(t, err)
+	require.Equal(t, ref1.PayloadCID, blk1.Cid())
+
+	ref2, err := exch.Index().GetRef(blk2.Cid())
+	require.NoError(t, err)
+	require.Equal(t, ref2.PayloadCID, blk2.Cid())
+
+	// check if bstore has blocks
+	has, err := exch.Index().Bstore().Has(blk1.Cid())
+	require.NoError(t, err)
+	require.Equal(t, true, has)
+
+	has, err = exch.Index().Bstore().Has(blk2.Cid())
+	require.NoError(t, err)
+	require.Equal(t, true, has)
+
+	// drop only ref1 (will tag corresponding blocks for eviction)
+	err = exch.Index().DropRef(blk1.Cid())
+	require.NoError(t, err)
+
+	// evict all tagged nodes
+	exch.Index().GC()
+
+	// check if GC did remove tagged block1 ...
+	has, err = exch.Index().Bstore().Has(blk1.Cid())
+	require.NoError(t, err)
+	require.Equal(t, false, has)
+
+	// ... but not block2
+	has, err = exch.Index().Bstore().Has(blk2.Cid())
+	require.NoError(t, err)
+	require.Equal(t, true, has)
```

Comment:
Can you add a testcase for when a block gets evicted? Good way to learn also how the eviction thresholds work etc.

Reply:
Regarding these tests:
- I'm not sure why you're creating full Exchange instances, these are `Index` unit tests meaning they should be testing the index in isolation. Can you just make an Index instance like for the other tests in this file?
- I feel like TestEvictBlock doesn't actually test the eviction and since tagging is already tested in TestGC, it should prob test real LFU eviction using bounds or just be removed.